### PR TITLE
RD-7117 Stage system tests failed on deployments_spec

### PR DESCRIPTION
**Pull request by leon-good-life**
## Description
Jira ticket: RD-7117

## Screenshots / Videos

Before:
Screenshot 2023-03-23 at 12 24 05

After:
Screenshot 2023-03-23 at 12 25 26

## Checklist

- [x] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [x] I followed the [UI Testing Guidelines](https://cloudifysource.atlassian.net/l/cp/udCVfvrx) and verified that all tests/checks have passed.
- [x] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [x] I added proper labels to this PR.

## Tests
`deployments_spec` - ✅ passed:
https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/4794/pipeline/

`deployments_spec` - ✅ passed:
https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/4793/pipeline

 

## Documentation
N/A

### CODE REVIEW

File: app/widgets/common/deployments/DeploymentActionsMenu.utils.ts [hunk 1]
```diff
@@ -4,7 +4,7 @@ import type { MenuItem } from './DeploymentActionsMenu.consts';
 import { permissions, actions } from './DeploymentActionsMenu.consts';
 
 function hasEnvironmentLabel(deploymentLabels: Label[]) {
-    return deploymentLabels.some(deploymentLabel => {
+    return deploymentLabels?.some(deploymentLabel => {
         return deploymentLabel.key === 'csys-obj-type' && deploymentLabel.value === 'environment';
     });
 }
```
Comment (Vorbert-Kruk) [inline app/widgets/common/deployments/DeploymentActionsMenu.utils.ts:10]: WDYT about the following modification?

```suggestion
function hasEnvironmentLabel(deploymentLabels: Label[] = []) {
    return deploymentLabels.some(deploymentLabel => {
        return deploymentLabel.key === 'csys-obj-type' && deploymentLabel.value === 'environment';
    });
}
```

File: app/widgets/common/deployments/DeploymentActionsMenu.utils.ts [hunk 2]
```diff
@@ -4,7 +4,7 @@ import type { MenuItem } from './DeploymentActionsMenu.consts';
 import { permissions, actions } from './DeploymentActionsMenu.consts';
 
 function hasEnvironmentLabel(deploymentLabels: Label[]) {
-    return deploymentLabels.some(deploymentLabel => {
+    return deploymentLabels?.some(deploymentLabel => {
```
Comment (qooban) [inline app/widgets/common/deployments/DeploymentActionsMenu.utils.ts:7]: What was the actual error? What was the value of `deploymentLabels` passed to `hasEnvironmentLabel` function?

According to the type - `deploymentLabels` are supposed to be an array of `Label` objects, so adding `?` seems to me like a way to confuse reader who would expect `deploymentLabels` to not be undefined/null.

File: app/widgets/common/deployments/DeploymentActionsMenu.utils.ts (hunk 1 above)
Comment (leon-good-life) [reply app/widgets/common/deployments/DeploymentActionsMenu.utils.ts]: [b6662f3](https://github.com/cloudify-cosmo/cloudify-stage/pull/2496/commits/b6662f38f932f90939227d8e6e09984dbe6cf7d5)

File: app/widgets/common/deployments/DeploymentActionsMenu.utils.ts (hunk 2 above)
Comment (leon-good-life) [reply app/widgets/common/deployments/DeploymentActionsMenu.utils.ts]: Here is the error:
Screenshot 2023-03-23 at 13 09 10

The value of `deploymentLabels` was `undefined` that passed to `hasEnvironmentLabel`:
Screenshot 2023-03-23 at 13 11 50

**Review by Vorbert-Kruk — APPROVED** [review]

**Comment by Vorbert-Kruk** [comment]
I've approved due to a change that is fixing the issue
But @qooban as `deploymentLabels` could not exist in the component, it would be beneficial if we'd adjust the typing used within component accordingly
I don't know if the current PR is a good place for that or if a new ticket should be created (maybe while working on ([RND-21](https://cloudifysource.atlassian.net/browse/RND-21)) I could tackle it)
But as the issue seems to have a deeper root cause, I've started to think about possible approaches to it
WDYT? 🤔

**Comment by leon-good-life** [comment]
I run the test (`deployments_spec`) again after addressing review comment and it passed - ✅ passed:
https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/4794/pipeline/

File: app/widgets/common/deployments/DeploymentActionsMenu.utils.ts (hunk 2 above)
Comment (qooban) [reply app/widgets/common/deployments/DeploymentActionsMenu.utils.ts]: Hmm... Interesting. Could you check what is the actual response from manager API for that problematic deployment? (on /deployments endpoint) Do we receive nullish value on `labels` field?

**Comment by leon-good-life** [comment]
I'm encountering 502 bad gateway when I'm installing version 7.0.0. So I unable to test it on 7.0.0. I guess the test is failing also on 7.0.0. So I will make a PR to `7.0.0-build`.

File: app/widgets/common/deployments/DeploymentActionsMenu.utils.ts (hunk 2 above)
Comment (qooban) [reply app/widgets/common/deployments/DeploymentActionsMenu.utils.ts]: @leon-good-life @Vorbert-Kruk 

I checked that by myself. Will share my findings here:

1. In this specific place in test we were using mocked data for deployments endpoint response. This mocked data lacked `labels` property. That was direct reason for the problem and it should be fixed in the first place.
2. When deployment does not have labels defined, it responds with `labels: []`, so there's no need to do any workarounds.

Having that said and knowing that Leon is on vacation, I'm taking over RD-7117 and I'll provide fixes for `master` and `7.0.0-build` branches.

**Pull request merged**